src/Grpc.AspNetCore.Server/Internal/ObjectMethodExecutor/AwaitableInfo.cs

```diff
@@ -0,0 +1,142 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal struct AwaitableInfo
```

Comment:
minor suggestion: `readonly struct`. It's immutable as the dodo anyway, and big enough to potentially benefit from by const ref.

Reply:
Could be interesting, might want to benchmark the difference @JamesNK

Reply:
Filed https://github.com/aspnet/AspNetCore/issues/7167